Form/Type/DocumentType.php

```diff
@@ -20,9 +20,24 @@
 use Doctrine\Bundle\PHPCRBundle\Form\ChoiceList\PhpcrOdmQueryBuilderLoader;
 use Symfony\Bridge\Doctrine\Form\ChoiceList\EntityLoaderInterface;
 use Symfony\Bridge\Doctrine\Form\Type\DoctrineType;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
 
 class DocumentType extends DoctrineType
 {
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        parent::setDefaultOptions($resolver);
+        $type = $this;
+        $loader = function (Options $options) use ($type) {
+            $qb = $options['query_builder'] ?: $options['em']->createQueryBuilder()->fromDocument($options['class'], 'a');
+
+            return $type->getLoader($options['em'], $qb, $options['class']);
```

Comment:
but it will get into 2.3? if we do not duplicate the method which versions of Symfony2 will not be supported? if we do keep this method, we should add a note about when we can remove the method.